stock_reserve_area/models/stock_move.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2023 ForgeFlow SL.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import OrderedSet, float_is_zero
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    area_reserved_availability = fields.Float(
+        string="Reserved in Area",
+        digits="Product Unit of Measure",
+        readonly=True,
+        copy=False,
+        help="Quantity that has been reserved in the reserve"
+        " Area of the source location.",
+    )
+    reserve_area_id = fields.Many2one(
+        "stock.reserve.area", compute="_compute_reserve_area_id", store=True
+    )
+    is_out_area_move = fields.Boolean(compute="_compute_is_out_area_move")
+
+    @api.constrains("reserve_area_id")
+    def check_reserve_area_id(self):
+        for rec in self:
+            if rec.picking_id and rec.picking_id.reserve_area_id != rec.reserve_area_id:
+                raise ValidationError(
+                    _(
+                        "Move Reservation Area can't be different from "
+                        "picking Reservation Area."
+                    )
+                )
+
+    @api.constrains("is_out_area_move")
+    def check_is_out_area_move(self):
+        for rec in self:
+            if (
+                rec.picking_id
+                and rec.picking_id.is_out_area_picking != rec.is_out_area_move
```

Comment:
An api.constrains with a condition in 2 computed (non-stored) fields, why? is this code even triggered?